### DISCUSSION TITLE
Use hooks with only director and  in-memory db

### DISCUSSION THREAD

**Discussion by kuznar** [sorry-cypress/sorry-cypress#719] [Ideas]
I want a way to know when all the tests of a ci-build-id are completed, and I want the results for things like pipeline gating. The hooks (https://github.com/sorry-cypress/sorry-cypress/tree/master/packages/director/src/lib/hooks/reporters) seem to do this.

Today if you wish to utilize the hooks (bitbucket, slack, web, etc) you must deploy and maintain the full project (dashboard, api, director, and a db). I would like to utilize the hook with just the director and it's in-memory db.

Propose editing hook init function to still start hooks even when using an in-memory db (or we can add and additional env var to turn it on in in-memory mode): https://github.com/sorry-cypress/sorry-cypress/blob/master/packages/director/src/lib/hooks/init.ts#L45 

But then we need a way to configure the hooks without the dashboard. Propose adding an endpoint to the director's express server to add the hooks object to projects.

**Comment by agoldis** [comment]
Good idea

**Reply by kuznar** [reply]
Awesome! Should have a PR out soon. Will be very open to critiques.

**Reply by kuznar** [reply]
@agoldis Do you have a rough idea for when the next release is?

**Reply by agoldis** [reply]
It's already release automatically on merge to main

**Reply by kuznar** [reply]
@agoldis Ah gotcha, I see things updated in "latest" but any idea about a release cut (2.6.0)?

**Comment by LWilliamson1** [comment]
@agoldis Is it possible to get a release cut. I’ve been watching a couple of issues and we can only pull tags in and not latest - fun bureaucracy issues.

**Comment by agoldis** [comment]
https://github.com/sorry-cypress/sorry-cypress/releases/tag/v2.5.2

On Wed, Mar 8, 2023 at 9:20 AM LWilliamson1 ***@***.***>
wrote:

> @agoldis <https://github.com/agoldis> Is it possible to get a release
> cut. I’ve been watching a couple of issues and we can only pull tags in and
> not latest - fun bureaucracy issues.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/sorry-cypress/sorry-cypress/discussions/719#discussioncomment-5244437>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMP4KHKNI4XBYVWDFDUHYLW3C5VTANCNFSM6AAAAAAT26JNM4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 
Andrew
currents.dev
